java/core/src/test/org/apache/orc/impl/TestStringHashTableDictionary.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.orc.impl;
+
+import org.apache.hadoop.io.Text;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class TestStringHashTableDictionary {
+
+  /**
+   * A extension for {@link StringHashTableDictionary} for testing purpose by overwriting the hash function.
+   *
+   */
+  private static class SimpleHashDictionary extends StringHashTableDictionary {
+    public SimpleHashDictionary(int initialCapacity) {
+      super(initialCapacity);
+    }
+
+    /**
+     * Obtain the prefix for each string as the hash value.
+     * All the string being used in this test suite will contains its hash value as the prefix for the string content.
+     * this way we know the order of the traverse() method.
+     */
+    @Override
+    int getIndex(Text text) {
```

Comment:
Shall we have a test where the actual HashFunction is used as well?

Reply:
Will address.